pkg/plugin/keyvault_test.go

```diff
@@ -133,16 +131,10 @@ func TestNewKeyVaultClient(t *testing.T) {
 		t.Run(test.desc, func(t *testing.T) {
 			kvClient, err := NewKeyVaultClient(test.config, test.vaultName, test.keyName, test.keyVersion, test.proxyMode, test.proxyAddress, test.proxyPort, test.managedHSM)
 			if err != nil {
-				t.Fatalf("newKeyVaultClient() failed with error: %v", err)
+				t.Fatalf("NewKeyVaultClient() failed with error: %v", err)
 			}
 			if kvClient == nil {
-				t.Fatalf("newKeyVaultClient() expected kv client to not be nil")
-			}
-			if !strings.Contains(kvClient.GetUserAgent(), "k8s-kms-keyvault") {
```

Comment:
@aramase I could not figure out what the point of these getters were in the public interface and why this was being tested